### PR TITLE
Standardize Token Program Validatation

### PR DESCRIPTION
**Pull request by coachchucksol**
All IXs that interact with the Token Program calls `load_token_program` which checks for the original SPL token id: `spl_token::id()`

There are weird side effects that need to be considered when implementing Token2022. My suggestion would be to hold off until we have time to consider the implications of Token2022. When we do, we should potentially create token_2022 versions of our IXs.

### CODE REVIEW

File: core/src/loader.rs [hunk 1]
```diff
@@ -206,10 +203,7 @@ pub fn load_token_account(
 /// # Returns
 /// * `Result<(), ProgramError>` - The result of the operation
 pub fn load_token_mint(info: &AccountInfo) -> Result<(), ProgramError> {
-    if !(info.owner.eq(&spl_token::id()) || info.owner.eq(&spl_token_2022::id())) {
```
Comment (buffalu) [inline core/src/loader.rs:209]: should we just check for spl token program here?

File: core/src/loader.rs [hunk 2]
```diff
@@ -118,10 +118,7 @@ pub fn load_associated_token_account(
     owner: &Pubkey,
     mint: &Pubkey,
 ) -> Result<(), ProgramError> {
-    if token_account.owner.ne(&spl_token::id()) {
```
Comment (buffalu) [inline core/src/loader.rs:121]: shouldn't this be left as is?

File: core/src/loader.rs (hunk 1 above)
Comment (buffalu) [reply core/src/loader.rs:209]: the description talks about removing Token2022, imo should keep the spl_token::id check here and remove the spl22 check

Comment (coachchucksol) [reply core/src/loader.rs:209]: agreed and reverted

File: core/src/loader.rs (hunk 2 above)
Comment (coachchucksol) [reply core/src/loader.rs:121]: It should, reverted

File: core/src/loader.rs [hunk 3]
```diff
@@ -161,11 +160,14 @@ pub fn load_token_account(
     token_account: &AccountInfo,
     owner: &Pubkey,
     mint: &Pubkey,
-    token_program_info: &AccountInfo,
+    token_program: &AccountInfo,
 ) -> Result<(), ProgramError> {
-    spl_token_2022::check_spl_token_program_account(token_account.owner)?;
+    if token_program.key.ne(&spl_token::id()) {
+        msg!("Account data is empty");
```
Comment (buffalu) [inline core/src/loader.rs:166]: wrong error message

File: vault_program/src/enqueue_withdrawal.rs [hunk 4]
```diff
@@ -56,9 +56,11 @@ pub fn process_enqueue_withdrawal(
     load_signer(staker, false)?;
     load_associated_token_account(staker_vrt_token_account, staker.key, &vault.vrt_mint)?;
     load_signer(base, false)?;
-    load_token_program(token_program)?;
     load_system_program(system_program)?;
 
+    // Only the original spl token program is allowed
+    load_token_program(token_program)?;
```
Comment (buffalu) [inline vault_program/src/enqueue_withdrawal.rs:62]: nit: put back into correct order

**Review by buffalu — APPROVED** [review]

File: core/src/loader.rs (hunk 3 above)
Comment (coachchucksol) [reply core/src/loader.rs:166]: Done

File: vault_program/src/enqueue_withdrawal.rs (hunk 4 above)
Comment (coachchucksol) [reply vault_program/src/enqueue_withdrawal.rs:62]: Done

**Pull request merged**